### ISSUE TITLE
minify js can cause XSS

### ISSUE THREAD

**Issue by dohvis** [wilsonzlin/minify-html#32]
```python
text="""<script> var a = '\\u003cscript\\u003ealert(1)\\u003c/script\\u003e' </script>"""
minify(text, minify_js=True)
```

```html
'<script>var a="<script>alert(1)</script>";</script>'
```

Even if the user's input is escaped with \\u003c, etc., the defense logic becomes useless because it is replaced again when minified. Is there any way to solve this?

**Comment by wilsonzlin** [comment]
This is an esbuild behaviour and should be raised with esbuild, as it would be much easier for esbuild to automatically emit `<\/script` for any literal substring `</script`. To see why, there are a few possible syntax combos that can cause an unintentional ending tag after minifying:

- `` `</script` `` (to prevent, add backslash before slash)
- `1 < /script/.exec(a).length` (add space before slash)
- `` `  ${`  ${a < /script/}  `}  ` `` (same as above)
- `// </script>`
- `/* </script>`

These require different code insertions to escape, but it's not really possible to distinguish between these via simple text replacement; a JS parser would be needed, which is why within esbuild is a better candidate for making these adjustments. However, in the meantime, I'm thinking of restoring d8f6bd as it works for the most common case (the one mentioned in this issue).

**Comment by dohvis** [comment]
@wilsonzlin Thanks for answering and suggestion. 

But `<script>` is user's input, so we cannot adjust such as `<script>` strings. Also we don't use esbuild.. just use https://github.com/pallets/jinja for rendering.

**Comment by wilsonzlin** [comment]
esbuild is the minifier included with and used by minify-html and is causing the aforementioned behaviour, which is why I mentioned raising with esbuild.

> But `<script>` is user's input, so we cannot adjust such as `<script>` strings.

I'm not sure what is meant by this, can you clarify this? Regardless, I released a new version `0.4.8` that escapes any `</script>` in string literals in JS code so that they don't get misinterpreted as the closing tag and allow HTML to be injected after the tag. It should solve this issue; if you are able to give it a test, let me know how it goes.
